model/group_request.go

```diff
@@ -53,9 +48,27 @@ func (request *CreateGroupRequest) Validate() error {
 	return nil
 }
 
+// NewCreateGroupRequestFromReader will create a CreateGroupRequest from an io.Reader with JSON data.
+func NewCreateGroupRequestFromReader(reader io.Reader) (*CreateGroupRequest, error) {
+	var createGroupRequest CreateGroupRequest
+	err := json.NewDecoder(reader).Decode(&createGroupRequest)
+	if err != nil && err != io.EOF {
+		return nil, errors.Wrap(err, "failed to decode create group request")
+	}
+
+	createGroupRequest.SetDefaults()
+	err = createGroupRequest.Validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid group create request")
+	}
+
+	return &createGroupRequest, nil
+}
+
 // PatchGroupRequest specifies the parameters for an updated group.
 type PatchGroupRequest struct {
 	ID            string
+	MaxRolling    *int64
```

Comment:
Why is this a pointer to an int64 instead of just an int64?

Reply:
For that patch logic it's cleaner/easier to use pointers(except for ID of course which is how the patch is correlated).

You can see how the other options are pointers as well.

As we know, the group patching logic is different from other updates (for reasons I can't remember or was never told). I plan to update everything else eventually to standardize.

Reply:
Do you mean the patch logic that appears starting on line 96 in `Apply`? Pardon me if I'm being thick, but the only difference I can see caused by using pointers in the `PatchGroupRequest` type definition is that now for each member, you have to check to see if it's `nil` before you check the value, which doesn't seem cleaner to me?

Can you tell me what I'm clearly not seeing? Sorry, again, if I'm just missing something obvious. If this is the current version of one of the assignments in `Apply`:

```go
	if p.Name != nil && *p.Name != group.Name {
		applied = true
		group.Name = *p.Name
	}
```

then if `p.Name` was a `string` instead of a `*string` then the patch would be
```go
if p.Name != group.Name {
    applied = true
    group.Name = p.Name
}
```
There must be something I'm not understanding.

Reply:
Or are we doing it like this because `nil` represents no change in a patch? 

In other words, and I don't know if this analogy helps, but if this was a functional language, `group.Name` in the above snipped might be type `Option[String]` of either type `Some[String]` or `None`?

Reply:
Yes, exactly. `nil` is way to "fall through" if no value is there and gives you another option beyond empty string.

We do this elsewhere like in the Mattermost server config. I think the intent was to have the nil check+patch logic for all types of cloud server object updates, but that was forgotten up till now.

Either way, yes, `nil` represents no change.